lib/BaseStrategy.ts

```diff
@@ -58,10 +67,7 @@ export abstract class BaseStrategy extends Strategy {
    */
   authenticate(req: Request): void {
     const token = this.extractToken(req);
-    if (!token) {
-      return this.fail('Missing credentials', 400);
-    }
-    const tokenPayload = this.decodeToken(token);
+    const siweObject = this.extractSiwe(req);
```

Comment:
@JGiter regarding your comment about error handling, does it make sense to have a try/catch here? Is this what you have in mind?
```suggestion
    let siweObject: SiweReqPayload | null;
    try {
      siweObject = this.extractSiwe(req);
    } catch(err) {
      return this.fail(err as string)
    }
```

Reply:
Yes, but it would also be useful to pass error, which would allow to identify error reason

Reply:
Right, makes sense. How about this?
```suggestion
    let siweObject: SiweReqPayload | null;
    try {
      siweObject = this.extractSiwe(req);
    } catch(err) {
      return this.error(new Error('error while extracting SIWE object', { cause: err }));
    }
```
I'm trying to use the `cause` prop of Error:
- https://stackoverflow.com/a/69799384
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause

Reply:
The message from `siwe` doesn't provide any useful details. The only thing we can do is to communicate that message is invalid in some way. To identify this case I propose to throw some easily recognizable errror type, say `InvalidSiweMessage`